### PR TITLE
core/rawdb,eth/protocols,p2p: prealloc slice size

### PR DESCRIPTION
**Pull request by tianyeyouyou**

### CODE REVIEW

**Comment by Mazzika1** [comment]
Ok

**Review by rjl493456442 — APPROVED** [review]

**Comment by Mazzika1** [comment]
Okay

في الجمعة، ٣١ مايو ٢٠٢٤, ٥:٥٦ ص rjl493456442 ***@***.***> كتب:

**Comment by tianyeyouyou** [comment]
@Mazzika1 please stop spamming.

File: p2p/peer.go [hunk 1]
```diff
@@ -511,9 +511,10 @@ type PeerInfo struct {
 // Info gathers and returns a collection of metadata known about a peer.
 func (p *Peer) Info() *PeerInfo {
 	// Gather the protocol capabilities
-	var caps []string
-	for _, cap := range p.Caps() {
-		caps = append(caps, cap.String())
+	peerCaps := p.Caps()
+	caps := make([]string, len(peerCaps))
+	for i, cap := range peerCaps {
+		caps[i] = cap.String()
```
Comment (karalabe) [inline p2p/peer.go:517]: Please revert this. We almost never call this method, so it's runtime is irrelevant. The original code was a lot nicer to read/understand.

File: core/rawdb/accessors_chain.go [hunk 2]
```diff
@@ -864,13 +864,13 @@ func ReadAllBadBlocks(db ethdb.Reader) []*types.Block {
...
-	var blocks []*types.Block
-	for _, bad := range badBlocks {
+	blocks := make([]*types.Block, len(badBlocks))
+	for i, bad := range badBlocks {
 		block := types.NewBlockWithHeader(bad.Header)
 		if bad.Body != nil {
 			block = block.WithBody(*bad.Body)
 		}
-		blocks = append(blocks, block)
+		blocks[i] = block
```
Comment (karalabe) [inline core/rawdb/accessors_chain.go:873]: Please revert this. This is run once in a blue moon, it's not epformance critical to make it less readable.

File: p2p/peer.go (hunk 1 above)
Comment (tianyeyouyou) [reply p2p/peer.go]: Done

File: core/rawdb/accessors_chain.go (hunk 2 above)
Comment (tianyeyouyou) [reply core/rawdb/accessors_chain.go]: Done

**Pull request merged**